metrics/amplitude.js

```diff
@@ -0,0 +1,332 @@
+  recoveryEmail: 'reset_password',
+  unblockCode: 'unblock',
+  verifyEmail: 'registration',
+  verifyLoginEmail: 'login',
+  verifyLoginCodeEmail: 'login',
+  verifyPrimaryEmail: 'verify',
+  verifySyncEmail: 'registration',
+  verifySecondaryEmail: 'secondary_email'
+}
+
+const NEWSLETTER_STATES = {
+  optIn: 'subscribed',
+  optOut: 'unsubscribed',
+};
+
+const EVENT_PROPERTIES = {
+  [GROUPS.activity]: NOP,
+  [GROUPS.connectDevice]: mapConnectDeviceFlow,
+  [GROUPS.email]: mapEmailType,
+  [GROUPS.emailFirst]: NOP,
+  [GROUPS.login]: NOP,
+  [GROUPS.registration]: NOP,
+  [GROUPS.settings]: mapDisconnectReason,
+  [GROUPS.sms]: NOP
+}
+
+function NOP () {}
+
+function mapConnectDeviceFlow (eventType, eventCategory, eventTarget) {
+  const connect_device_flow = CONNECT_DEVICE_FLOWS[eventCategory];
+
+  if (connect_device_flow) {
+    const result = { connect_device_flow };
+
+    if (eventTarget) {
+      result.connect_device_os = eventTarget;
+    }
+
+    return result;
+  }
+}
+
+function mapEmailType (eventType, eventCategory, eventTarget, data) {
+  const email_type = EMAIL_TYPES[eventCategory];
+
+  if (email_type) {
+    const result = { email_type, email_provider: data.emailDomain };
+
+    const { templateVersion } = data;
+    if (templateVersion) {
+      result.email_template = eventCategory;
+      result.email_version = templateVersion;
+    }
+
+    return result;
+  }
+}
+
+function mapDisconnectReason (eventType, eventCategory) {
+  if (eventType === 'disconnect_device' && eventCategory) {
+    return { reason: eventCategory };
+  }
+}
+
+module.exports = {
+  GROUPS,
+
+  EMAIL_TYPES,
+
+  /**
+   * Initialize an amplitude event mapper. You can read more about the amplitude
+   * event structure here:
+   *
+   *   https://amplitude.zendesk.com/hc/en-us/articles/204771828-HTTP-API
+   *
+   * And you can see our event taxonomy here:
+   *
+   *   https://docs.google.com/spreadsheets/d/1G_8OJGOxeWXdGJ1Ugmykk33Zsl-qAQL05CONSeD4Uz4
+   *
+   * @param {Object} services An object of client-id:service-name mappings.
+   *
+   * @param {Object} events   An object of name:definition event mappings, where
+   *                          each defintion value is itself an object with `group`
+   *                          and `event` string properties.
+   *
+   * @param {Map} fuzzyEvents A map of regex:definition event mappings. Each regex
+   *                          key may include up to two capturing groups. The first
+   *                          group is used as the `eventCategory` and the second is
+   *                          used as the `eventTarget`. Again each definition value
+   *                          is an object containing `group` and `event` properties
+   *                          but here `group` can be a string or a function. If it's
+   *                          a function, it will be passed the matched `eventCategory`
+   *                          as its argument and should return the group string.
```

Comment:
Can you leave a note that this returns a function, which is the actual mapper? I was confused how this function was confused until I looked at how this function is [invoked from the content server](https://github.com/mozilla/fxa-content-server/pull/6022/files#diff-2a83e43540a70e8aa80daa6c26a3de6dR92).